src/main/java/games/strategy/engine/framework/startup/ui/editors/EmailSenderEditor.java

```diff
@@ -188,6 +189,10 @@ private void testEmail() {
     }).start();
   }
 
+  private static String truncate(String text, int maxLength) {
```

Comment:
Might be able to use Guava's [`Ascii#truncate()`](http://google.github.io/guava/releases/23.0/api/docs/com/google/common/base/Ascii.html#truncate-java.lang.CharSequence-int-java.lang.String-) instead.  The docs say it is not safe for general Unicode text because of surrogate pairs, but it's basically the same implementation as below (i.e. we will split surrogate pairs, if present, just as well).

There are a few other locations where very similar string truncation code is present.  Maybe we should just replace them all as part of a separate PR?

Reply:
I cribbed the truncate function from the one other location, so made sense to fix both. 3c393a5fc
Good to know where the guava version is, I was looking for the apache util 'truncate' and had no luck finding that.

Reply:
The droid you're looking for is [`StringUtils#abbreviate()`](https://commons.apache.org/proper/commons-lang/javadocs/api-3.6/org/apache/commons/lang3/StringUtils.html#abbreviate-java.lang.String-int-).  Unfortunately, Apache Commons Lang is not currently in our dependency list.